akka-docs/src/main/paradox/stream/operators/Source-or-Flow/mapWithResource.md

```diff
@@ -0,0 +1,40 @@
+# mapWithResource
+
+Transform each stream element with the help of a resource.
+
+@ref[Simple operators](../index.md#simple-operators)
+
+## Signature
+
+@apidoc[Flow.mapWithResource](Flow) { scala="#mapWithResource%5BS%2C%20T%5D%28create%3A%20%28%29%20%3D%3E%20S%29%28f%3A%20%28S%2C%20Out%29%20%3D%3E%20T%2C%20close%3A%20S%20%3D%3E%20Option%5BT%5D%29%3A%20Repr%5BT%5D" java="#mapWithResource(akka.japi.function.Creator,akka.japi.function.Function2,akka.japi.function.Function)" }
+
+## Description
+
+Transform each stream element with the help of a resource.
+
```

Comment:
```suggestion
The functions are by default called on Akka's dispatcher for blocking IO to avoid interfering with other stream operations. 
See @ref:[Blocking Needs Careful Management](../../../typed/dispatchers.md#blocking-needs-careful-management) for an explanation on why this is important.
```